File: src/java/tusan/nodes/expression/Condition.java
package net.tutla.tums.tusan.nodes.expression;

import net.tutla.tums.tusan.Node;
import net.tutla.tums.tusan.lexer.Token;
import net.tutla.tums.tusan.lexer.TokenType;

import java.util.Objects;

public class Condition extends Node {
    public Boolean value;
    private Boolean opposite = false;

    public Condition(Token token){
        super(token);
        if (token.type == TokenType.LOGIC && token.value.equals("not")){
            this.opposite = true;
        }

    }

    public Condition create(){
        value = true;
        Expression expr1 = new Expression(token).create();
        if (interpreter.getNextToken().type == TokenType.LOGIC){
            String operator = interpreter.nextToken().value;
            Expression expr2 = new Expression(interpreter.nextToken()).create();
            if (operator.equals("and") || operator.equals("&&")){
                if ((Boolean) expr1.value && (Boolean) expr2.value){
                    value = true;
                } else {
                    value = false;
                }
            } else if (operator.equals("or") || operator.equals("||")) {
                if ((Boolean) expr1.value || (Boolean) expr2.value){
                    value = true;
                } else {
                    value = false;
                }
            } else if (operator.equals("contains")) {
                // WIP
            } else if (operator.equals("in")) {
                // WIP
            }
        } else {
            value = (Boolean) expr1.value;
        }

        if (opposite){
            if (value){
                value = false;
            } else {
                value = true;
            }
        }

        return this;
    }
}

File: src/python/src/tusan/nodes/expression/Condition.py
from tusk.node import Node
from tusk.token import Token

class ConditionNode(Node):
    def __init__(self,token:Token):
        self.interpreter = token.interpreter
        self.auto_eval = True
        self.token = token

        self.opposite = False

        if token.type == "LOGIC" and token.value=="not":
            self.opposite = True
            token = self.interpreter.next_token()
        
        self.token = token

    async def create(self):
        from tusk.nodes.expressions import ExpressionNode
        tkn1 = await ExpressionNode(self.token).create()
        if self.interpreter.get_next_token().type == "LOGIC":
            operator = self.interpreter.next_token()
            tkn2 = await ExpressionNode(self.interpreter.next_token()).create()
            if operator.value == "and" or operator.value == "&":
                if tkn1.value == True and tkn2.value == True: self.value = True
                else: self.value = False  
            elif operator.value == "or" or operator.value == "|":
                if tkn1.value == True or tkn2.value == True: self.value = True
                else: self.value = False  
            elif operator.value == "contains":
                if tkn2.value in tkn1.value: self.value = True
                else: self.value = False
            elif operator.value == "in":
                if tkn1.value in tkn2.value: self.value = True
                else: self.value = False
        else:
            self.value = bool(tkn1.value)
            self.type="1en"

        if self.opposite: 
            if self.value == True: self.value = False
            else: self.value = True

        

        return self